public/app/features/alerting/unified/components/rule-editor/QueryRows.tsx

```diff
@@ -156,12 +156,18 @@ export class QueryRows extends PureComponent<Props> {
               <div ref={provided.innerRef} {...provided.droppableProps}>
                 <Stack direction="column">
                   {queries.map((query, index) => {
+                    const isCondition = this.props.condition === query.refId;
                     const data: PanelData = this.props.data?.[query.refId] ?? {
                       series: [],
                       state: LoadingState.NotStarted,
                     };
                     const dsSettings = this.getDataSourceSettings(query);
-                    const error = data ? errorFromPreviewData(data) : undefined;
+                    let error: Error | undefined = undefined;
+                    if (data && isCondition) {
```

Comment:
Not entirely sure about this part because we give precedence to custom errors from the FE, but perhaps good enough here since the user has to resolve these first and any BE errors will show up when the user chooses another condition.